src/throttler.guard.ts

```diff
@@ -72,28 +72,40 @@ export class ThrottlerGuard implements CanActivate {
         }
       }
     }
-    const tracker = this.getTracker(req);
-    const key = this.generateKey(context, tracker);
-    const ttls = await this.storageService.getRecord(key);
-    const nearestExpiryTime = ttls.length > 0 ? Math.ceil((ttls[0] - Date.now()) / 1000) : 0;
+    const t = this.getTracker(req);
+    const trackers = Array.isArray(t) ? t : [t];
+    let maxTtlsLen = 0;
+    let nearestExpiryTime = 0;
+    const keys = []
+    for (const tracker of trackers) {
+      const key = this.generateKey(context, tracker);
+      const ttls = await this.storageService.getRecord(key);
```

Comment:
`await` inside a `for` loop doesn't seems good to me. Can't we use `.map`+`Promise.all`? :thinking:

Reply:
You are right, I revised in a new commit